### PR TITLE
create input component

### LINKED ISSUE #47: Input component
### Description

We need a generic Inputfield (`<input />`) styled to match our design system.

Figma:
https://www.figma.com/file/wnBveAG2ikUspFsQwM3GNE/Altinn-Studio-Apps?node-id=12857%3A8186

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by Magnusrm**
Create input component using design system

Screen Recording 2022-07-07 at 12 20 08

image

## Issue
fixes #47 Input component

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
- [x] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [x] All tests run green

### CODE REVIEW

File: src/components/TextField/TextField.module.css [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+.input-wrapper--default:hover {
+  border-color: var(--component-input-color-border-hover);
+}
+
+.input-wrapper--default:focus-within {
+  border-width: var(--component-input-border-width-focus);
+  border-color: var(--component-input-color-border-focus);
+  outline: none;
+  margin: 0px;
```
Comment (haakemon) [inline src/components/TextField/TextField.module.css:33]: nitpick;
Dont need units for 0 values :) 

```suggestion
  margin: 0;
```

**Review by haakemon — CHANGES_REQUESTED** [review]

File: src/components/TextField/TextField.module.css [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+}
+
+.input-wrapper--readonly-confirm .input-wrapper__field {
+  background-color: #D4F9E4;
+}
+
+.input-wrapper__icon--error {
+  padding-right: var(--component-input-border-width-default);
+  padding-left: var(--component-input-border-width-default);
+  min-width: 22px;
```
[the remark this answers is not in the document]
Comment (Magnusrm) [reply src/components/TextField/TextField.module.css:90]: Ser bra ut i Storybook, men knakk i appen. 😢 
image image

[the remark this answers is not in the document]
Comment (haakemon) [reply src/components/TextField/TextField.module.css:90]: Sikker på at du har fått med siste endring? Størrelsen ser også forskjellig ut på selve inputfeltet

**Review by haakemon — APPROVED** [review]

**Comment by haakemon** [comment]
Design system v2 is setting 

```css
font-size: inherit;
line-height: inherit;
```
which computes to 16px for font-size and 24px for line-height. This causes the inputfields to be much bigger when introduced in app-frontend repo. We need to look into how to deal with this, as the inputfields should be 36px tall. Currently they will be 40px instead because of the inherited values.

**Review by haakemon — CHANGES_REQUESTED** [review]
As mentioned previously, we need to fix the height issues

File: README.md [hunk 3]
```diff
@@ -35,6 +36,10 @@ Lint checks and auto-fixes will be run automatically on commit.
 
 New components can be added by executing `yarn add-component <ComponentName>`. The name of the component should be written using PascalCase. This will generate all important files for you in the correct location, and also update the index file for exporting the component. The generated code includes some `TODO` statements that you should fix.
 
+#### Adding new dependencies
+
+When adding new dependencies, you should also add that dependency to the `external` array in `rollup.config.js`. This is done to avoid having the dependency being part of the bundle.
```
Comment (olemartinorg) [inline README.md:41]: Can't we just reference `dependencies` from `package.json` in this file instead? That's the way I've solved that problem before. I'll create a separate issue for it.

Comment (olemartinorg) [reply README.md:41]: Added comment in #62 instead

**Review by olemartinorg — APPROVED** [review]
Looks good to me! :tada: Of course, the mentioned size issue should be looked at.

File: README.md (hunk 3 above)
Comment (haakemon) [reply README.md:41]: That seems like a great idea - lets follow up in 62.

**Comment by haakemon** [comment]
Height issue has been partially fixed. It looks good when the component is used in app-frontend, but it doesnt look correct when viewed in Storybook. 

Created a separate task to follow up this issue #76

**Review by haakemon — APPROVED** [review]

**Pull request merged**